tests/test_pricing_model.py

```diff
@@ -0,0 +1,1279 @@
+                    # Combining the without_fee and the fee, we calculate with_fee as:
+                    #   with_fee = 102.50516899477225 - 0.5013436672973448 = 102.0038253274749
+                    102.0038253274749,
+                    # We set up the problem as:
+                    #   100_100 ** (1 - T) + (300_000 - d_y) ** (1 - T) = k
+                    #
+                    # Solving for d_y, we get the following calculation:
+                    #   d_y = 300_000 - (k - 100_100 ** (1 - T)) ** (1 / (1 - T)) = 102.50516899477225
+                    #
+                    # Note that this is slightly smaller than the without slippage value
+                    102.50516899477225,
+                    # Since we are buying bonds, in_ is an amount of base and we calculate the fee using the spot price as:
+                    #   fee = 0.2 * (p - 1) * 100 = 0.5013436672973448
+                    0.5013436672973448,
+                ),
+            ),
+            # Medium slippage trade - in_ is 10% of share reserves.
+            (
+                TestCaseCalcOutGivenInSuccess(
+                    in_=10_000,
+                    share_reserves=100_000,
+                    bond_reserves=100_000,
+                    token_out="pt",
+                    fee_percent=0.01,
+                    days_remaining=182.5,
+                    time_stretch_apy=0.05,
+                    share_price=1,
+                    init_share_price=1,
+                ),
+                # From the input, we have the following values:
+                # - T = 0.02253584403159705
+                # - p = 1.0250671833648672
+                # - k = 302_929.51067963685
+                (
+                    # Using the spot price, the expected output without slippage or fees is given by:
+                    #   1.0250671833648672 * 10_000 = 10250.671833648672
+                    10250.671833648672,
+                    # Combining the without_fee and the fee, we calculate with_fee as:
+                    #   with_fee = 10235.514826394327 - 2.506718336486724 = 10233.00810805784
+                    10233.00810805784,
+                    # We set up the problem as:
+                    #   110_000 ** (1 - T) + (300_000 - d_y) ** (1 - T) = k
+                    #
+                    # Solving for d_y, we get the following calculation:
+                    #   d_y = 300_000 - (k - 110_000 ** (1 - T)) ** (1 / (1 - T)) = 10235.514826394327
+                    #
+                    # Note that this is smaller than the without slippage value
+                    10235.514826394327,
+                    # Since we are buying bonds, in_ is an amount of base and we calculate the fee using the spot price as:
+                    #   fee = 0.01 * (p - 1) * 10_000 = 2.506718336486724
+                    2.506718336486724,
+                ),
+            ),
+            # TODO: The slippage should arguably be much higher. This is something
```

Comment:
might consider moving this comment into the pricing model code and referencing the test, otherwise we might lose track of it.